src/main/java/com/minecolonies/coremod/client/gui/WindowHutGuardTower.java

```diff
@@ -8,256 +8,480 @@
+        if (task.equals(GuardTask.PATROL))
+        {
+            patrolList.setDataProvider(new ScrollingList.DataProvider()
+            {
+                @Override
+                public int getElementCount()
+                {
+                    return patrolTargets.size();
+                }
+
+                @Override
+                public void updateElement(final int index, @NotNull final Pane rowPane)
+                {
+                    final BlockPos pos = patrolTargets.get(index);
+                    rowPane.findPaneOfTypeByID("position", Label.class).setLabelText(pos.getX() + " " + pos.getY() + " " + pos.getZ());
+                }
+            });
+        }
+        else if (task.equals(GuardTask.GUARD))
+        {
+            patrolList.setDataProvider(new ScrollingList.DataProvider()
+            {
+                @Override
+                public int getElementCount()
+                {
+                    return 1;
+                }
+
+                @Override
+                public void updateElement(final int index, @NotNull final Pane rowPane)
+                {
+                    final BlockPos pos = building.getGuardPos();
+                    rowPane.findPaneOfTypeByID("position", Label.class).setLabelText(pos.getX() + " " + pos.getY() + " " + pos.getZ());
+                }
+            });
+        }
+
+        ScrollingList mobsList = findPaneOfTypeByID(GUI_ELEMENT_LIST_MOBS, ScrollingList.class);
+        mobsList.setDataProvider(new ScrollingList.DataProvider()
+        {
+            @Override
+            public int getElementCount()
+            {
+                return mobsToAttack.size();
+            }
+
+            @Override
+            public void updateElement(final int index, final Pane rowPane)
+            {
+                final String name = mobsToAttack.get(index).getName();
 
-        registerButton(BUTTON_TASK_PATROL, this::switchTask);
-        registerButton(BUTTON_TASK_FOLLOW, this::switchTask);
-        registerButton(BUTTON_TASK_GUARD, this::switchTask);
-        registerButton(BUTTON_SET_TARGET, this::setTarget);
+                rowPane.findPaneOfTypeByID(GUI_LIST_ELEMENT_NAME, Label.class).setLabelText(name);
 
-        buttonTaskPatrol = this.findPaneOfTypeByID(BUTTON_TASK_PATROL, Button.class);
-        buttonTaskFollow = this.findPaneOfTypeByID(BUTTON_TASK_FOLLOW, Button.class);
-        buttonTaskGuard = this.findPaneOfTypeByID(BUTTON_TASK_GUARD, Button.class);
+                final Button switchButton = rowPane.findPaneOfTypeByID(GUI_LIST_BUTTON_SWITCH, Button.class);
 
-        buttonSetTarget = this.findPaneOfTypeByID(BUTTON_SET_TARGET, Button.class);
+                if (mobsToAttack.get(index).getAttack())
+                {
+                    switchButton.setLabel(GUI_SWITCH_ON);
+                }
+                else
+                {
+                    switchButton.setLabel(GUI_SWITCH_OFF);
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Use already-defined constant 'BUTTON_PATROL_MODE' instead of duplicating its value here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1192)